### PR TITLE
Plumbing for BelosBlockCGSolMgr so that Assert Positive Definiteness flag gets passed to BelosCGIter

### PR DESCRIPTION
**Pull request by DABH**
@trilinos/belos

## Motivation

This was brought up as part of https://github.com/trilinos/Trilinos/issues/6005 .  I tried to do something like `belosList.set("Assert Positive Definiteness", false);` in my code while setting up a BelosBlockCGSolMgr but the value didn't get passed through to the underlying BelosCGIter.  AFACIT the plumbing was jut missing -- not sure if there was a good reason for that?  If not, this PR adds the necessary plumbing so the flag value is now correctly passed through.

https://github.com/trilinos/Trilinos/issues/6005

### CODE REVIEW

File: packages/belos/src/BelosBlockCGSolMgr.hpp [hunk 1]
```diff
@@ -828,6 +838,8 @@ BlockCGSolMgr<ScalarType,MV,OP,true>::getValidParameters() const
       "The string to use as a prefix for the timer labels.");
     pl->set("Orthogonalization", static_cast<const char *>(orthoType_default_),
       "The type of orthogonalization to use: DGKS, ICGS, or IMGS.");
+    pl->set("Assert Positive Definiteness",static_cast<bool>(assertPositiveDefiniteness_default_),
```
Comment (mhoemmen) [inline packages/belos/src/BelosBlockCGSolMgr.hpp]: It's a `bool`; you don't need the `static_cast`.

Comment (DABH) [reply packages/belos/src/BelosBlockCGSolMgr.hpp]: I think you're right -- any idea why, in that block of code, there are several static casts of things to ints and bools?  Even for the static casts to const char*, those variables that are being casted seem to already be of those types.  Are any of these static casts here actually necessary?

Comment (mhoemmen) [reply packages/belos/src/BelosBlockCGSolMgr.hpp]: @DABH The casts are probably not necessary.  Belos is nearly 15 years old -- it's possible that some compilers we had to support back that had trouble deducing some of those types.  It's also likely that later developers went overboard with the casts.  I'll be happy to approve PRs that purge these; I just don't have time at the moment to clean this up.

Comment (mhoemmen) [reply packages/belos/src/BelosBlockCGSolMgr.hpp]: btw I'm also not convinced that ParameterList needs overloads for `const char value[]` and `char value[]`; the former should suffice.  [This example](https://godbolt.org/z/hXjTcB) compiles even with C++98.

**Pull request merged**